crates/matrix-sdk-ui/tests/integration/timeline/mod.rs

```diff
@@ -507,3 +507,98 @@ async fn test_duplicate_maintains_correct_order() {
     let content = items[3].as_event().unwrap().content().as_message().unwrap().body();
     assert_eq!(content, "C");
 }
+
+#[async_test]
+async fn test_pin_event() {
```

Comment:
If I understand correctly, this test checks 2 things:

1. When I try to unpin something that is not pinned I receive `false`
2. When I pin something I am allowed to pin I receive `true`

I would prefer a few separate tests, preferably each with very simple Given-When-Then structure (i.e. set up, do something, check results). I often actually write sentences starting with "Given", "When" and "Then" as comments in my test. I find it makes it much easier to read the test and find the part that is actually being tested.

I'd also like to check a little more about what actually happened.

So I might write tests like:

* `pinning_an_event_makes_it_appear_in_the_pinned_list`
* `unpinning_an_event_makes_it_disappear_from_the_pinned_list`
* `trying_to_unpin_a_nonexistent_event_returns_false`
* `trying_to_unpin_an_unpinned_event_returns_false`
* `trying_to_pin_an_event_without_permission_is_an_error`
* `trying_to_unpin_an_event_without_permission_is_an_error`

To avoid repeating code in the tests, I would make the setup code use a function or even a `TestSetup` struct if needed.

I know this is my very opinionated way of writing tests :-) I will defend it enthusiastically at the pub, but will not insist you do it this way in this PR if you are not persuaded.